financial/present_value.py

```diff
@@ -1,38 +1,69 @@
 """
-Reference: https://www.investopedia.com/terms/p/presentvalue.asp
+Reference: https://www.auditexcel.co.za/blog/discounting-cash-flows-with-multiple-discount-rates/
 
 An algorithm that calculates the present value of a stream of yearly cash flows given...
-1. The discount rate (as a decimal, not a percent)
+1. An array containing tuples of discount rates and their associated duration.
+
+For example [(0.05, 2), (0.06, 3)] would mean that a discount rate of 5% is applied to
+the first 2 cash flows, while a discount rate of 6% is applied to the next 3 cash flows.
+
+If the discount rate is fixed for the entire duration, then the user may provide just
+the discount rate(as a float).
+
 2. An array of cash flows, with the index of the cash flow being the associated year
 
 Note: This algorithm assumes that cash flows are paid at the end of the specified year
 """
 
 
-def present_value(discount_rate: float, cash_flows: list[float]) -> float:
+def present_value(discount_rates: list[tuple], cash_flows: list[float]) -> float:
     """
     >>> present_value(0.13, [10, 20.70, -293, 297])
```

Comment:
I am confused as to why line 21 delivers `4.69` in the old code and `4.15` in the new.
Will it do the same if `0.13` is changed to `[0.13]` on line 21.
I would like us to keep the old tests (with modifications if required) to prove that the old and new are equivelent.

Reply:
- I am confused as to why line 21 delivers 4.69 in the old code and 4.15 in the new.
The formula produces 4.15, i have no idea why it was 4.69 in the first place. You can cross check with the formula found [here](https://en.wikipedia.org/wiki/Present_value). Another test came up with wrong results, fixed it as well.

-  Will it do the same if 0.13 is changed to [0.13] on line 21?
no it will produce an error. The idea is if there is a fixed rate, the user need only provide a float number of the rate, and if its a floating rate, the user provides a list of tuples.There is no instance where a variable of type `[float]` is provided.

Reply:
@cclauss the `enumerate()` function starts counting from `0` where `i` in 
`cash_flow / ((1 + discount_rate) ** i) for i, cash_flow in enumerate(cash_flows)` should start counting from 1, not 0, hence the computational error in the previous iteration. My implementation produces the correct result.

Reply:
What does `enumerate(cash_flows, 1)` do?

Reply:
Should
`def present_value(discount_rates: list[tuple]` be changed to
`def present_value(discount_rates: float | list[tuple]`?

Reply:
> Should `def present_value(discount_rates: list[tuple]` be changed to `def present_value(discount_rates: float | list[tuple]`?

Yes that's better, changing it rn

Reply:
> What does `enumerate(cash_flows, 1)` do?

It fixes the problem but i think that's beside the point. The test cases in place had different results than my tests because the previous implementation calculated wrong results. My implementation provides the correct results which is why there were differences between the tests.

Reply:
@cclauss do i resolve the conversation or do i leave this up to reviewers? I'd appreciate a review at this time